trickle/trickle_server.go

```diff
@@ -435,18 +438,18 @@ func (s *Stream) getForRead(idx int) (*Segment, int, bool) {
 		return seg != nil && seg.idx == i
 	}
 	if idx == -1 {
-		idx = s.latestWrite
+		idx = s.nextWrite
 	}
 	segmentPos := idx % maxSegmentsPerStream
 	segment := s.segments[segmentPos]
-	if !exists(segment, idx) && (idx == s.latestWrite+1 || (idx == 0 && s.latestWrite == 0)) {
+	if !exists(segment, idx) && (idx == s.nextWrite || (s.nextWrite == 0 && idx == 1)) {
```

Comment:
should the last condition be generified to `idx > s.nextWrite`? (then i guess we'd need to reset `nextWrite`?)